src/Controller/Component/AuthComponent.php

```diff
@@ -597,7 +597,7 @@ public function deny($actions = null)
     public function setUser(array $user)
     {
         $this->session->renew();
-        $this->session->write($this->sessionKey, $user);
+        $this->session->write($this->sessionKey . '.user', $user);
```

Comment:
It shouldn't. user data should be accessed via the provided method

http://book.cakephp.org/2.0/en/core-libraries/components/authentication.html#accessing-the-logged-in-user